File: Algorithms/Strings/Mars Exploration.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'marsExploration' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts STRING s as parameter.
     */

    public static int marsExploration(string s)
    {
        int count = 0;
        for (int i = 0; i < s.Length; i+=3)
        {
            if (s[i] != 'S') count += 1;
            if (s[i+1] != 'O') count += 1;
            if (s[i+2] != 'S') count += 1;
        }
        return count;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string s = Console.ReadLine();

        int result = Result.marsExploration(s);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Strings/Mars Exploration.php
<?php

/*
 * Complete the 'marsExploration' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts STRING s as parameter.
 */

function marsExploration($s) {
    # Write your code here
    $count = 0;
    for ($i = 0; $i < strlen($s); $i+=3) {
        if ($s[$i] != 'S') $count += 1;
        if ($s[$i+1] != 'O') $count += 1;
        if ($s[$i+2] != 'S') $count += 1;
    }
    return $count;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$s = rtrim(fgets(STDIN), "\r\n");
$result = marsExploration($s);

fwrite($fptr, $result . "\n");
fclose($fptr);
